### PR TITLE
Gabbi/wrangler include exclude

### PR DESCRIPTION
**Pull request by gabbifish**
This PR updates the wrangler reference page to include the new `include/exclude` fields for wrangler's sites settings, as well as adds a new docs page about wrangler ignore functionality.

This PR also has additional content that exists in master but not in staging yet.

### CODE REVIEW

File: content/sites/ignore-assets.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+---
+title: Ignoring Subsets of Static Assets
+alwaysopen: true
+weight: 5
+---
+
+Worker Sites require [Wrangler](https://github.com/cloudflare/wrangler) - make sure to be on the [latest version](/quickstart/#updating-the-cli) - and the Workers [Unlimited plan](https://workers.cloudflare.com/sites#plans).
```
Comment (EverlastingBugstopper) [inline content/sites/ignore-assets.md]: Workers Sites

File: content/sites/ignore-assets.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+There are cases where users may not want to upload certain static assets to their Workers Sites.
+In this case, Workers Sites can also be configured to ignore certain files or directories using logic
```
Comment (EverlastingBugstopper) [inline content/sites/ignore-assets.md]: Is there a reason to have line breaks after each of these sentences?

File: content/sites/ignore-assets.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+There are cases where users may not want to upload certain static assets to their Workers Sites.
+In this case, Workers Sites can also be configured to ignore certain files or directories using logic 
+similar to [Cargo's optional include and exclude fields](https://doc.rust-lang.org/cargo/reference/manifest.html#the-exclude-and-include-fields-optional).
+This means that we use gitignore semantics when declaring which directory entries to include or ignore in uploads.
+
+### Exclusively including files/directories
+
+If you want to include only a certain set of files or directories in your `bucket`, you can add an `include` field to your
+`[site]` section of `wrangler.toml`:
+```toml
```
Comment (EverlastingBugstopper) [inline content/sites/ignore-assets.md]: nit: should add new lines before and after the toml section

File: content/sites/ignore-assets.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+entry-point = "workers-site" 
+include = ["included_dir"] # must be an array.
+```
+Wrangler will only upload files or directories matching the patterns in the `include` array.
+
+### Excluding files/directories
+
+If you want to exclude files or directories in your `bucket`, you can add an `exclude` field to your
+`[site]` section of `wrangler.toml`:
+```toml
```
Comment (EverlastingBugstopper) [inline content/sites/ignore-assets.md]: nit: should add new lines before and after the toml section

File: content/sites/ignore-assets.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+[site]
+bucket = "./public"
+entry-point = "workers-site" 
+exclude = ["excluded_dir"] # must be an array.
+```
+Wrangler will ignore files or directories matching the patterns in the `exclude` array when uploading assets to Wrangler.
```
Comment (EverlastingBugstopper) [inline content/sites/ignore-assets.md]: Wrangler is not uploading assets to Wrangler, it is uploading assets to Workers KV

File: content/sites/ignore-assets.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+### Include > Exclude
+
+If you provide both `include` and `exclude` fields, the `include` field will be used and the `exclude` field will be ignored.
```
Comment (EverlastingBugstopper) [inline content/sites/ignore-assets.md]: Is this what we want? I think I missed this part of the implementation and it probably makes sense/is convention but... I can see a world where we have an `include: ["dir/img"]` and an `exclude: ["dir/img/way-too-big.jpg"]` and you'd wanna use that.

File: content/sites/reference.md [hunk 7]
```diff
@@ -29,10 +29,12 @@ Worker Sites require the latest version of [Wrangler](https://github.com/cloudfl
...
-| ------------- | ---------------------------------------------------------------------------------- | -------------------------------- |
-| `bucket`      | The directory containing your static assets, path relative to your `wrangler.toml` | `bucket = "./public"`              |
-| `entry-point` | The location of your Worker script, default is `workers-site`                      | `entry-point = "./workers-site"` |
+| Key           | Value                                                                              | Example                          | Required |
+| ------------- | ---------------------------------------------------------------------------------- | -------------------------------- | -------- |
+| `bucket`      | The directory containing your static assets, path relative to your `wrangler.toml` | `bucket = "./public"`            | yes      |
+| `entry-point` | The location of your Worker script, default is `workers-site`                      | `entry-point = "./workers-site"` | yes      |
+| `include`     | A list of gitignore-style patterns for files or directories in `bucket` you exclusively want to upload. | `include = ["upload_dir"]` | no |
```
Comment (EverlastingBugstopper) [inline content/sites/reference.md]: maybe we should add a link to the include/exclude section of the docs here for more info

File: content/sites/ignore-assets.md (hunk 1 above)
Comment (gabbifish) [reply content/sites/ignore-assets.md]: ?

File: content/sites/ignore-assets.md (hunk 2 above)
Comment (gabbifish) [reply content/sites/ignore-assets.md]: Just makes it more readable in text editors; markdown will format without these newlines.

File: content/sites/ignore-assets.md (hunk 5 above)
Comment (gabbifish) [reply content/sites/ignore-assets.md]: oop yeah!

File: content/sites/ignore-assets.md (hunk 2 above)
Comment (EverlastingBugstopper) [reply content/sites/ignore-assets.md]: cool

File: content/sites/ignore-assets.md (hunk 6 above)
Comment (gabbifish) [reply content/sites/ignore-assets.md]: what you're suggesting is not consistent with cargo `include/exclude` behavior; the behavior above is!

File: content/sites/ignore-assets.md (hunk 1 above)
Comment (EverlastingBugstopper) [reply content/sites/ignore-assets.md]: not Worker Sites

File: content/sites/ignore-assets.md (hunk 6 above)
Comment (EverlastingBugstopper) [reply content/sites/ignore-assets.md]: cool :)

File: content/sites/ignore-assets.md (hunk 1 above)
Comment (gabbifish) [reply content/sites/ignore-assets.md]: oh wait we should fix this for all documentation pages then, they all say `Worker Sites`

**Pull request merged**